### ISSUE TITLE
Version 1.1.5 not working in Linux/WSL

### ISSUE THREAD

**Issue by gabrielmdesidera** [Azure/static-web-apps-cli#776]
**Describe the bug**
When running the command `swa deploy`, an error occurs during the deploy. This error occurs only in version 1.1.5 in Linux/WSL. The same SWA CLI version works fine in Windows, and versions 1.1.4 and below work fine in Linux/WSL. 

**To Reproduce**

1. Install version 1.1.5: `npm install -g @azure/static-web-apps-cli@1.1.5`
2. Type in command 'swa deploy ...'
3. See error

```
Welcome to Azure Static Web Apps CLI (1.1.5)

⠋ Preparing deployment. Please wait...(node:468) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) 
will be removed. Use fs.rm(path, { recursive: true }) instead
(Use `node --trace-deprecation ...` to show where the warning was created)
✖ An unknown exception has occurred
```

**Expected behavior**
```
Welcome to Azure Static Web Apps CLI (1.1.4)

✔ Project deployed to https://xxxxxxxxxxx.azurestaticapps.net 🚀
```

**Screenshots**
Azure Pipelines:
![image (7)](https://github.com/Azure/static-web-apps-cli/assets/98892170/f3f39d12-1eff-480a-9cad-f23c959abf91)

**Desktop (please complete the following information):**

- WSL: Ubuntu 22.04.1 LTS, using Node v18.0.0
- Azure Pipelines: Ubuntu 20.04, not sure which version of Node

**Comment by JesseDeH** [comment]
+1 also getting this error when deploying from a Linux agent.

**Comment by jancimajek** [comment]
+1 Also getting the same error when deploying from Linux runner (using Octopus deploy, although I don't think that's relevant)

**Comment by jimmystridh** [comment]
Also getting the same error on both MacOS and Linux and no way of running an older version (!).

**Comment by jimmystridh** [comment]
for anyone needing a workaround I published my repo locally, created a new gitrepo and pushed the static build to that, created a new static web app and used this git repo, got a deployment token for my actual static web app using `swa deploy --print-token` and then switched the repo secret to this new key and re-ran the github action.

Did this in a bit of a rush so there's hopefully a less convoluted way to deploy with the broken swa cli out there. Directly using the docker container in https://github.com/Azure/static-web-apps-deploy was on the list as well.

**Comment by gabrielmdesidera** [comment]
@jimmystridh Earlier today I managed to downgrade to version 1.1.4 by running `npm install -g @azure/static-web-apps-cli@1.1.4`. Then the `swa deploy` command worked.

**Comment by jimmystridh** [comment]
> @jimmystridh Earlier today I managed to downgrade to version 1.1.4 by running `npm install -g @azure/static-web-apps-cli@1.1.4`. Then the `swa deploy` command worked.

interesting, will give that a try in case I need to deploy again before this is fixed, thanks!

**Comment by Gaffen** [comment]
Thanks for the temp fix @jimmystridh

**Comment by nimro** [comment]
Seeing this too on Ubuntu 22.04.3. Can confirm downgrading to 1.1.4 is a valid workaround, thanks @gabrielmdesidera

**Comment by spzm** [comment]
I confirm the same issue on MacOS 14 (Sonoma) with 1.1.5, rollback to 1.1.4 fixed the issue. thanks @gabrielmdesidera

**Comment by nimro** [comment]
Worse, the `swa` cli exits `0` (success) after printing this error 🤦
![image](https://github.com/Azure/static-web-apps-cli/assets/867976/e7be75d5-2631-45ff-8015-bdbe8ccf80ba)

**Comment by jimmystridh** [comment]
> Worse, the `swa` cli exits `0` (success) after printing this error 🤦 ![image](https://user-images.githubusercontent.com/867976/281379348-7f320c14-ea5a-47cc-a6a0-f1d13bdf9d9e.png)

Yup, our CI builds were passing so this went unnoticed for a while until it was noticed the frontend wasn't up-to-date, not great.

**Comment by cjk7989** [comment]
Hello all,
We have identified and confirmed the issue on Linux. It is caused by the child_process.spawn() function, which swa-cli uses to execute the StaticSiteClient binary. This function behaves differently on Windows and Linux. We are working on a fix and will update as soon as possible.
Thank you for your advices and apologise for the inconvenience.

**Comment by cjk7989** [comment]
> Hello all, We have identified and confirmed the issue on Linux. It is caused by the child_process.spawn() function, which swa-cli uses to execute the StaticSiteClient binary. This function behaves differently on Windows and Linux. We are working on a fix and will update as soon as possible. Thank you for your advices and apologise for the inconvenience.

We have released version 1.1.7. Thanks for your patience and support, and please let me know if there are any other problems.
